.circleci/config.yml

```diff
@@ -3,91 +3,158 @@
 
 version: 2.1
 
+orbs:
+  codecov: codecov/codecov@3.2.5
+  macos: circleci/macos@2
+
 # Workflows orchestrate a set of jobs to be run;
 workflows:
-  version: 2
-  build-test-deploy:
+  build-test:
     jobs:
-      - build-and-test
+      - validate-code
+      - test-ios:
+          requires:
+            - validate-code
+      - test-tvos:
+          requires:
+            - validate-code
+      - build_xcframework_and_app:
+          requires:
+            - validate-code
+
+commands: 
+  install_dependencies:
+    steps:
+      # restore pods related caches
+      - restore_cache:
+          name: Restoring Gemfile Cache
+          keys:
+            - 1-gems-{{ checksum "Gemfile.lock" }}
 
-jobs:
-  build-and-test:
-    macos:
-      xcode: 13.0.0 # Specify the Xcode version to use
+      # make sure we're on the right version of cocoapods
+      - run:
+          name: Verify Cocoapods Version
+          command: bundle check || bundle install --path vendor/bundle
 
-    steps:
-      - checkout
+      # save cocoapods version gem data
+      - save_cache:
+          name: Saving Gemfile Cache
+          key: 1-gems-{{ checksum "Gemfile.lock" }}
+          paths:
+            - vendor/bundle
 
       # restore pods related caches
       - restore_cache:
+          name: Restoring CocoaPods Cache
           keys:
             - cocoapods-cache-v5-{{ arch }}-{{ .Branch }}-{{ checksum "Podfile.lock" }}
             - cocoapods-cache-v5-{{ arch }}-{{ .Branch }}
             - cocoapods-cache-v5
-            - 1-gems-{{ checksum "Gemfile.lock" }}
-
-      # install Gemfile
-      - run: bundle check || bundle install --path vendor/bundle --clean
 
       # install CocoaPods - using default CocoaPods version, not the bundle
       - run:
           name: Repo Update & Install CocoaPods
-          command: bundle exec pod install --repo-update
+          command: make ci-pod-install
 
       # save pods related files
       - save_cache:
           name: Saving CocoaPods Cache
           key: cocoapods-cache-v5-{{ arch }}-{{ .Branch }}-{{ checksum "Podfile.lock" }}
           paths:
             - ./Pods
+            - ./SampleApps/TestApp/Pods
```

Comment:
Need to remove the `./SampleApp/TestApp` path.

Reply:
Need to also rotate the cache version to "v6" since we removed "vendor/bundle" and "SampleApps/TestApp/Pods"